code/cfile/cfilesystem.cpp

```diff
@@ -1202,11 +1418,31 @@ CFileLocationExt cf_find_file_location_ext( const char *filename, const int ext_
 			search_order[num_search_dirs++] = i;
 	}
 
-	memset( longname, 0, sizeof(longname) );
-	memset( filespec, 0, sizeof(filespec) );
+	// fixup filename and sub directory path, if needed
+	SCP_string filespec = filename;
+	SCP_string sub_path;
+
+	normalize_directory_separators(filespec);
+
+	auto seperator = filespec.rfind(DIR_SEPARATOR_CHAR);
+
+	if (seperator != SCP_string::npos) {
+		sub_path = filespec.substr(0, seperator);
+		sub_path += DIR_SEPARATOR_CHAR;
+
+		filespec.erase(0, seperator+1);
+	}
 
 	// strip any existing extension
-	strncpy(filespec, filename, MAX_FILENAME_LEN-1);
+	// (NOTE: to be fully retail compatible, we need to support multiple periods for something like *_1.5.wav,
+	//        which means that we need to strip a length of >2 only, assuming that all valid ext are at least 2 chars)
+	auto dot = filespec.find_last_of(".");
```

Comment:
For consistencies sake with all the other "find last char" methods used in this file, this probably should be an rfind too. And search for a char literal, not a string literal

Reply:
Yep, that's my fault. I did that in a couple of other places too for some reason but went back and fixed them. Obviously missed one.